### PR TITLE
Instrument with ActiveSupport::Notifications

### PR DESCRIPTION
**Pull request by danielmorrison**
Taking some inspiration from https://github.com/ruby-grape/grape/pull/1086

This may change a bit, but want to give us something to start with. 

I'm also adding more tests along the way.

### CODE REVIEW

File: lib/twirp/rails/handler.rb [hunk 1]
```diff
@@ -31,8 +31,14 @@ def process(name, request, env)
...
       # which is *not* necessarily the same as the action name.
       def process_action(name)
-        run_callbacks(:process_action) do
-          send_action(name)
+        ActiveSupport::Notifications.instrument("handler_run_callbacks.twirp_rails", handler: self.class.name, action: action_name, env: @env, request: @request) do
+          run_callbacks(:process_action) do
+            ActiveSupport::Notifications.instrument("handler_run.twirp_rails", handler: self.class.name, action: action_name, env: @env, request: @request) do |payload|
+              response = send_action(name)
+              payload[:response] = response
+              response
```
Comment (albus522) [inline lib/twirp/rails/handler.rb:39]: In ruby `=` will always return the passed in value
```suggestion
              payload[:response] = send_action(name)
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -14,6 +14,7 @@ jobs:
     strategy:
       matrix:
         ruby:
+          - '2.7.8'
```
Comment (albus522) [inline .github/workflows/ci.yml:17]: Add 3.0 as well?

Comment (danielmorrison) [reply .github/workflows/ci.yml:17]: yeah, I guess I should care 🤣

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -14,8 +14,10 @@ jobs:
     strategy:
       matrix:
         ruby:
-          - '3.1.3'
-          - '3.2.0'
+          - 2.7
+          - '3.0' # keep as string or it turns into "3" which pulls the newest 3.x, not 3.0.x
```
Comment (danielmorrison) [inline .github/workflows/ci.yml:18]: TIL

**Pull request merged**